File: du-backend/src/main/java/com/dwin/du/api/entity/PersonProduct.java
package com.dwin.du.api.entity;
import jakarta.persistence.*;
import lombok.*;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "person_product")
public class PersonProduct {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @ManyToOne
    @JoinColumn(name = "product_id")
    private Product product;

    @ManyToOne
    @JoinColumn(name = "person_id")
    private Person person;

    @Column(name= "part_of_price")
    private double partOfPrice;

    @Column(name = "quantity")
    private int quantity;

    @Column(name = "is_compensation")
    private boolean compensation;

    @Column(name = "is_settled")
    private  boolean settled;
}

File: divvyup/DivvyUp.Shared/Models/PersonProduct.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace DivvyUp_Shared.Models
{
    public class PersonProduct
    {
        [Column("id")]
        public int Id { get; set; }

        [Required]
        [ForeignKey("Person")]
        [Column("person_id")]
        public int PersonId { get; set; }
        public Person Person { get; set; }

        [Required]
        [ForeignKey("Product")]
        [Column("product_id")]
        public int ProductId { get; set; }
        public Product Product { get; set; }

        [Column("part_of_price")]
        public decimal PartOfPrice { get; set; }

        [Column("quantity")]
        public int Quantity { get; set; }

        [Column("is_compensation")]
        public bool Compensation { get; set; }

        [Column("is_settled")]
        public bool Settled { get; set; }
    }
}
